app/Http/Controllers/CanteensController.php

```diff
@@ -41,8 +41,24 @@ public function byWeek(Request $request)
         $start = $date->startOfWeek()->toDateString().' 00:00:00';
         $end = $date->endOfWeek()->toDateString().' 23:59:59';
 
-        $canteens = Canteens::with('menus')->whereBetween('date', [$start, $end])->orderBy('date');
+        $canteens_query = Canteens::with('menus')->whereBetween('date', [$start, $end])->orderBy('date');
 
-        return self::_after($request, $canteens, null);
+        // order canteens by date
+
+        $canteens = $canteens_query->get()->toArray();
+        $maxDate = new Carbon();
+        if (0 !== count($canteens)) {
+            $maxDate = $canteens_query->latest('updated_at')->first()->updated_at;
+        }
+        $canteens_sorted = array();
+        foreach ($canteens as $canteen) {
+            usort($canteen['menus'], function ($item1, $item2) {
+                return $item1['type'] <=> $item2['type'];
```

Comment:
The task to do may have been ambiguous, so sorry for that, we have to order the `$canteen['menus']` by type given the correct order is `Soup`, `Main dish`, `Dessert`.

Reply:
The `type` field can have values `0` for soup, `1` for main dish and `2` for desserts, so we can order simply by the value of `type`.